File: api/entity/container.go
package entity

import (
	"time"

	"github.com/gusantoniassi/navegante/core/entity"
)

type Container struct {
	ID         entity.ContainerID `json:"id"`
	Cmd        string             `json:"cmd"`
	Entrypoint string             `json:"entrypoint"`
	Created    time.Time          `json:"created"`
	Name       string             `json:"name"`
	State      string             `json:"state"`
	Status     string             `json:"status"`

	Image      string            `json:"image"`
	Ports      []string          `json:"ports"`
	Labels     map[string]string `json:"labels"`
	Volumes    []string          `json:"volumes"`
	Networks   []string          `json:"networks"`
	Statistics *entity.Stat      `json:"statistics"`
}

func NewContainer(c *entity.Container) Container {
	portMappings := make([]string, len(c.Ports))
	volumes := make([]string, len(c.Volumes))
	networks := make([]string, len(c.Networks))

	for i, p := range c.Ports {
		portMappings[i] = p.String()
	}

	for i, v := range c.Volumes {
		volumes[i] = v.String()
	}

	for i, n := range c.Networks {
		networks[i] = n.String()
	}

	newC := Container{
		ID:         c.ID,
		Cmd:        c.Cmd.String(),
		Entrypoint: c.Entrypoint.String(),
		Created:    c.Created,
		Name:       c.Name,
		State:      c.State,
		Status:     c.Status,
		Image:      c.Image.String(),
		Ports:      portMappings,
		Volumes:    volumes,
		Networks:   networks,
	}

	return newC
}

File: web/core/lib/entity/Container.ts
import { Stat } from './Stat';

export class Container {
  constructor(obj: any) {
    this.id = obj.id;
    this.cmd = obj.cmd;
    this.entrypoint = obj.entrypoint;
    this.created = new Date(obj.created);
    this.name = obj.name;
    this.state = obj.state;
    this.status = obj.status;
    this.image = obj.image;
    this.ports = obj.ports;
    this.labels = obj.labels;
    this.volumes = obj.volumes;
    this.networks = obj.networks;
    this.statistics = new Stat(obj.statistics);
  }

  id: string

  cmd: string

  entrypoint: string

  created: Date

  name: string

  state: string

  status: string

  image: string

  ports: Array<string>

  labels: Array<object>

  volumes: Array<string>

  networks: Array<string>

  statistics: Stat
}
